### DISCUSSION TITLE
Swell add to cart inside a next.js application ... Stock level handling not working?

### DISCUSSION THREAD

**Discussion by UZuccaro** [swellstores/community#85] [Help] [answered]
Hi,
I am developing an e-commerce based on next.js using the Swell Frontend API (only to handle add-to-cart and cart itself).
I can add products to the cart (without any problem), but it seems that the stock_level check (that the API is supposed to do) is not made after every add-to-cart, allowing me to add more products to the cart than I could add (exceeding the stock level).

Here's an example, using a test product (it's a variation):

![Schermata 2023-01-13 alle 09 37 41](https://user-images.githubusercontent.com/5860049/212275542-529194af-8cda-4b7a-b472-0efcac3e00bf.png)

As you can see I added three times this product variation to the cart (check the "quantity" field for this cart item).
But the stock lever for this variation it's 1 (as you can see inside the variant.stock_level data).

How is this possibile?
Where am I messing up?

Thanks for you support.

**Comment by greph** [comment, marked as answer]
Hi @UZuccaro , currently the stock check is performed during the cart conversion process and not during add to cart. It's a known feature gap in on the front-end. As a workaround, the stock level can be validated against on the front-end, and those items can be shown as unavailable or a user-friendly message shown.

**Reply by osseonews** [reply]
Please fix this feature gap. It is very difficult and error prone to validate stock level on the front end. Slows down the entire app. The stock check should be not be performed during cart conversion, as this is already too late in the checkout process, and again causes alot of issues from a usability standpoint. Thanks.

**Reply by greph** [reply]
Thanks @osseonews for the feedback. I've passed this along to our product team.
